src/security/builtin_plugins/authentication/src/authentication.c

```diff
@@ -1221,7 +1216,7 @@ DDS_Security_ValidationResult_t begin_handshake_request(dds_security_authenticat
   handshake_message->class_id = ddsrt_strdup(AUTH_HANDSHAKE_REQUEST_TOKEN_ID);
   handshake_message->properties._length = 0;
   handshake_message->properties._buffer = NULL;
-  handshake_message->binary_properties._length = 8;
+  handshake_message->binary_properties._length = tsz;
```

Comment:
`tsz == idx` at this point, is it not? One should at least add an assertion that this is the case. Perhaps just use `idx`. Furthermore, `tsz` is a bit cryptic and `idx` a bit generic. I'd rather seem them renamed to `tokcount` and `tokidx`, or something.